File: models/Flag.py
from sqlalchemy import Column, Integer, String
from utils.database import Base

class Flag(Base):
    __tablename__ = "flags"

    id = Column(Integer, primary_key=True, index=True)
    flagString = Column(String(255), unique=True, index=True, nullable=False)
    assignedTeam = Column(String(100), nullable=True)
    containerId = Column(String(100), nullable=True)

File: models/Flag.js
import { Sequelize, DataTypes, Model } from 'sequelize';
import { sequelize } from '../utils/database.js';

class Flag extends Model {}

Flag.init({
    flagString: {
        type: DataTypes.STRING,
        allowNull: false,
    },
    containerId: {
        type: DataTypes.STRING,
        allowNull: false,
    },
    assignedTeam: {
        type: DataTypes.STRING,
        allowNull: true, // kalau null berarti untuk semua team
    },
    status: {
        type: DataTypes.ENUM('active', 'solved'),
        defaultValue: 'active',
        allowNull: false,
    }
},{
    sequelize,
    modelName: 'Flag'
})

export default Flag;